src/commands/NFD.ts

```diff
@@ -0,0 +1,1003 @@
+          embeds: [embed],
+          files: [imageAttachment],
+          ephemeral: silent,
+        })
+      })
+      .catch((err) => {
+        console.error('something went very wrong in making a collection', err)
+      })
+  }
+
+  @Slash('gift', { description: 'Gift your NFD to another chatter. How kind.' })
+  @SlashGroup('nfd')
+  async gift(
+    @SlashOption('nfd', {
+      type: ApplicationCommandOptionType.String,
+      description: 'The name of the NFD to be gifted.',
+      required: true,
+    })
+    nfd: string,
+    @SlashOption('recipient', {
+      type: ApplicationCommandOptionType.User,
+      description: 'The chatter to receive the NFD.',
+      required: true,
+    })
+    recipient: User | GuildMember,
+    interaction: CommandInteraction
+  ) {
+    return await this.performGift(nfd, recipient, false, interaction)
+  }
+
+  // Function that actually carries out the transaction
+  private async performGift(
+    nfd: string,
+    recipient: User | GuildMember,
+    sudo: boolean,
+    interaction: CommandInteraction
+  ) {
+    // Confirm the caller isn't on cooldown (sudo overrides)
+    if (!sudo) {
+      const caller = await this.client.nFDEnjoyer.findUnique({ where: { id: interaction.user.id } })
+      if (!caller) {
+        return interaction.reply({
+          content: 'The dinochain is broken. The calling user is missing :(',
+          ephemeral: true,
+        })
+      }
+
+      if (caller.lastGiftGiven.getTime() + this.GIFT_COOLDOWN > Date.now()) {
+        return interaction.reply({
+          content: `You're gifting too often. You can gift again in <t:${Math.round(
+            (caller.lastGiftGiven.getTime() + this.GIFT_COOLDOWN) / 1000
+          )}:R>.`,
+          ephemeral: true,
+        })
+      }
+      // and confirm the caller isn't gifting to themselves (sudo overrides)
+      if (recipient.id == interaction.user.id) {
+        return interaction.reply({ content: "You can't gift something to yourself.", ephemeral: true })
+      }
+    }
+
+    // Now confirm the NFD exists
+    const nfd_item = await this.getNFDByName(nfd)
+    if (!nfd_item) {
+      return interaction.reply({ content: "I couldn't find an NFD with that name.", ephemeral: true })
+    }
+
+    // Confirm that the caller owns the NFD (sudo overrides)
+    if (nfd_item.owner != interaction.user.id && !sudo) {
+      return interaction.reply({ content: "You can't gift something you don't own!", ephemeral: true })
+    }
+
+    // All checks have passed. Carry out the change of owner.
+    const ownerList = nfd_item.previousOwners + `,<@${recipient.id}>`
+    await this.client.nFDItem.update({
+      where: {
+        name: nfd_item.name,
+      },
+      data: {
+        previousOwners: ownerList,
+        owner: recipient.id,
+      },
+    })
+
+    if (sudo) {
+      return interaction.reply({
+        content: `${interaction.user} reassigned ${nfd_item.name} to ${recipient} using their mod powers.`,
```

Comment:
We want to avoid pings when possible. Probably a good idea to put this here and in the non mod version

```suggestion
        content: `${interaction.user} reassigned ${nfd_item.name} to ${recipient} using their mod powers.`,
        allowedMentions: { repliedUser: false }
```